framework/src/main/java/org/checkerframework/framework/type/TypeFromMemberVisitor.java

```diff
@@ -72,10 +75,30 @@ public AnnotatedTypeMirror visitMethod(MethodTree node, AnnotatedTypeFactory f)
         result.setElement(elt);
 
         ElementAnnotationApplier.apply(result, elt, f);
+        new TypeVarAnnotator().visit(result, f);
```

Comment:
It seems a bit strange that this is only used in `visitMethod`.
Could you add a test case outside of a method, e.g. in a field initializer or in a superclass instantiation?
In the original code, all type variables were checked. Now only ones in method declarations are checked. Or am I missing where else this is used?

Reply:
All uses of type parameters must be in the class that declared them (including in an enclosing class).  For all uses, except those in a method declaration, the type of the type variable is computed using `TypeFromTree#fromTypeTree`.  `org.checkerframework.framework.type.TypeFromTypeTreeVisitor#forTypeVariable` is called for all uses of type variables and that method looks up the annotations on type parameters and type parameter upper bounds from the tree.  Types in method signatures aren't created using `TypeFromTree#fromTypeTree`, so it needs special handling.  (I tried calling `TypeFromTree#fromTypeTree` on all the type trees in a method signature, but it caused the framework to crash because recursive type variables were not initialized as expected.)   (I did add a test case for fields.)

My main goal with this pull request is to move this code out of `InheritedFromClassAnnotator`, so that I can remove and replace that class in a later pull request.  Since the code in `InheritedFromClassAnnotator#visitTypeVariable` has no documentation and was added in the initial Checker Framework commit, I'm just guessing.   If you don't find the explanation in the above paragraph convincing, then I'm ok with moving this code so that it applies to all type variables again.  (Though, I'm not really sure where that would be.)